source/GameWindow.cpp

```diff
@@ -108,6 +149,12 @@ bool GameWindow::Init()
 	// Settings that must be declared before the window creation.
 	Uint32 flags = SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE | SDL_WINDOW_ALLOW_HIGHDPI;
 
+#ifdef __linux__
+	// Force fullscreen for steam deck
+	if(IsSteamDeck())
+		Preferences::Set("fullscreen", true);
```

Comment:
```suggestion
		flags |= SDL_WINDOW_FULLSCREEN_DESKTOP;
```

I find setting the preference just so that the next if becomes true a bit unnecessary; might as well do the correct thing immediately.

Reply:
Fullscreen preference looks like it does a few things.  I considered doing your suggestion but decided to set the preference since that was the intent.  Are you sure we want to do it this way?

Reply:
Also, take for example flags for both fullscreen and maximized getting set.  That would render the following if/else with both code paths logically. I think setting the preference is safer to avoid unknown behavior.

Reply:
Thinking about it some more, yeah makes sense. :+1: